public/app/features/plugins/sql/components/QueryHeader.tsx

```diff
@@ -123,10 +134,15 @@ export function QueryHeader({
               transparent={true}
               showLabel={true}
               value={queryRowFilter.filter}
-              onChange={(ev) =>
+              onChange={(ev) => {
```

Comment:
sorry, i'd recommend slightly reworking these sections, i understand that they previously had the `ev.target instanceof HTMLInputElement &&` check, but it now makes it hard to read.
what do you think about something like:
```ts
(ev) => {
                if (!(ev.target instanceof HTMLInputElement)) {
                  return;
                }

                reportInteraction('grafana_sql_filter_toggled', {
                  datasource: query.datasource?.type,
                  displayed: ev.target.checked,
                });

                onQueryRowChange({ ...queryRowFilter, filter: ev.target.checked });
              }}
```

basically, we do the instanceof-check at the beginning only once... this could be applied to all of them.